docs/extract.rst

```diff
@@ -1,51 +1,51 @@
-===============================================================================
-Background
-===============================================================================
-
-Filing
-^^^^^^
-
-A single XBRL filing or instance contains a list of facts and a list of contexts.
-A fact is considered to be an atomic unit of data. It contains a single value, a
-name, data type, and a reference to a context. The context provides information relevant
-to interpret the fact. A context always contains a time period and entity that the
-pertains to, as well as any other "dimensions" that can be defined to further
-identify the fact in question.
-
-Taxonomy
-^^^^^^^^
-
-An XBRL taxonomy provides important metadata and structure for interpreting
-facts reported in a filing. All facts that can be reported for a given taxonomy have
-a Concept defined in the taxonomy. Concepts contain metadata that provide a description,
-units, and other helpful information. Not all concepts are facts, however, they can also
-represent a group of facts and provide information about that group.
-
-To group facts into "fact tables", a taxonomy defines relationships between concepts.
-These relationships are defined in a Link Role. Each link role contains a Directed
-Acyclic Graph (DAG) of concepts. Concepts that correspond to individual facts will
-be the "leaf nodes" in this graph.
-
 ===============================================================================
 Parsing
 ===============================================================================
 
 Taxonomy Parsing
 ^^^^^^^^^^^^^^^^
-`Arelle <https://arelle.org/arelle/>`__ is a popular open source library for working with XBRL
-data, and it is used for parsing the taxonomy. After parsing the taxonomy, each link
-role or fact table is used to generate a table schema for a SQL table. This is done
-by traversing down the concept DAG to find leaf nodes, and create a column for each
-leaf node. After this process is complete, we can parse individual filings, and
-populate these constructed tables with the data in the filing.
+`Arelle <https://arelle.org/arelle/>`__ is a popular open source library for working
+with XBRL data, and it is used for parsing the taxonomy. Arelle can parse a taxonomy
+from either a URL to the taxonomy entry point, or a local path to a zipfile containing
+the entire taxonomy.
+
+The extractor will first take a parsed taxonomy and construct a Taxonomy object defined
+in ``ferc_xbrl_extractor/taxonomy.py``. This is done because the data structures used by
+Arelle are not well documented, so it's immediately move to costum structures. After
+creating these structures, the extractor will generate
+`frictionless datapackage <https://specs.frictionlessdata.io/data-package/>`__, which
+contains a schema for the new SQLite DB, as well as useful metadata.
+
+Each table in the SQLite DB is derived from a Link Role in the Taxonomy. This is done
```

Comment:
I think linking to the glossary from the terms in this document would be helpful. Sphinx has [built-in support](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-glossary) term references / glossaries.

Reply:
Thanks! I tried to do linking, but I wasn't aware of the glossary directive that makes this much easier